Code/Unsupervised_reconstruction/Source/dynamicPrograms.py

```diff
@@ -3,13 +3,13 @@
 import torch
 from torch import Tensor
 from models.probcache import ProbCache
-from source.editModel import EditModel
+from Source.editModel import EditModel
 from models.types import InferenceData_SamplesEmbeddings, InferenceData_Cognates
 from typing import Union
 
 BIG_NEG = -1e9
 
-
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
It was added quickly to carry out sampling tests, but deserves further thought, in particular to ensure that vectors are created on the right device.

TDLR; Currently ok, but in the future should be removed and the initialisation of vectors on the chosen device should be standardised.